### ISSUE TITLE
Should sp-with-modes affect sp--get-pair-list-context?

### ISSUE THREAD

**Issue by expez** [Fuco1/smartparens#417]
This snippet is from the default config:

```elisp
(sp-with-modes '(emacs-lisp-mode
                 inferior-emacs-lisp-mode
                 lisp-interaction-mode
                 lisp-mode)
  (sp-local-pair "'" nil :actions nil)
  (sp-local-pair "`" "'" :when '(sp-in-string-p) :actions '(insert wrap)))
```

It says to do nothing with the pair `''`, because of the quote character.  

In `sp-region-ok-p` I use `(sp--get-opening-regexp (sp--get-pair-list-context))` to do some work but that function still returns `'` as an opening pair.  Is this as it should be?  How can I get a filtered list of pairs?

**Comment by Fuco1** [comment]
I think what you want is `(sp--get-allowed-pair-list)`... these functions are a historic mess of crap. After I fix the bugs caused by the current input handling rewrite, this is the next thing to clean up.

**Comment by expez** [comment]
Thanks, `sp--get-allowed-regexp`, which builds on `sp--get-allowed-pair-list` seems perfect.

**Comment by Fuco1** [comment]
I consider these functions implementation details, if you need it we should add a new primitive that gives you the data you want.

So, what is it you use these for?

**Comment by expez** [comment]
I'm not attempting to use this outside of `smartparens`, but I did find a situation that `sp-region-ok` doesn't handle.  I trimmed the end of the region for whitespace, before trying to walk it with `sp-forward-sexp` but I should really trim it for anything which can't affect the validity of region.  I noticed this in javascript when I tried to delete a line and the semicolon at the end caused the region being acted upon to shrink.
